src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const currentState = Object.assign({}, state);
+  const modifiedStates = actions.map(action => {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          currentState[key] = action.extraData[key];
+        }
+
+        const withAdd = Object.assign({}, currentState);
+
+        return withAdd;
```

Comment:
Don't repeat yourself: you can move this logic after switch/case instead of doing this in every case